### PR TITLE
bug: update activities to complete on PR merge.

### PR DESCRIPTION
**Pull request by markawm**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

This is basically copying the workaround for issue #2410 from `PromoteOptions.waitForGitOpsPullRequest()` into
the workflow controller so that PipelineActivities get updated when the promotion PR merges. Normally, this is
done via `jx promote` during the build, but if that fails/times out then this ensures the workflow controller
tidies up the status to reflect that the PR is merged & promotion is complete.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

**Comment by markawm** [comment]
I've duplicated some of the code from `jx promote` into the workflow controller, where I see _largely_ the same logic, but tbh I'm not sure why we have the same (or very very similar) code in both places: both:
 - Can these be merged? Is it just a refactoring that hasn't happened yet?
 - why are both the build job (via `jx promote`) and the workflow controller taking on what looks to me like essentially the same role in promotion?

**Comment by jstrachan** [comment]
@markawm refactoring to share code would be good. The workflow controller came second; the longer term idea is to move more of the logic for promotion to a single controller and have less happen in the pipelines themselves; though we've yet to totally go 100% in on the workflow controller right now. We may be able to use bits of the workflow controller triggered via knative build pipeline at some point too to simplify the code + logic

File: pkg/jx/cmd/controller_workflow.go [hunk 1]
```diff
@@ -507,6 +519,9 @@ func (o *ControllerWorkflowOptions) pollGitStatusforPipeline(activity *v1.Pipeli
 			log.Warnf("Failed to get PR number: %s", err)
 			return
 		}
+		if prNumber == 42 {
+			fmt.Printf("!!!Stop\n")
+		}
```
Comment (amuniz) [inline pkg/jx/cmd/controller_workflow.go]: wat?

Comment (markawm) [reply pkg/jx/cmd/controller_workflow.go]: D'oh - this was from when I was testing out the fix. But it got merged before I had a chance to remove it! :-(
See https://github.com/jenkins-x/jx/pull/3053

**Pull request merged**